sdk/js/index.ts

```diff
@@ -272,32 +283,43 @@ export default class StreamManager {
   }
 
   onStreamCreated(
-    handleStream: (stream: Stream) => null,
+    handleStream: (stream: Stream) => void,
     creator?: Address
   ): void {
-    const onLogs = (logs: Log[]) => {
-      logs
-        .map(
-          // Log is StreamCreated
-          (log: Log) =>
-            new Stream(
-              this,
-              log.topics[2] as Address, // creator
-              Number(log.topics[3]), // streamId
-              log.topics[4] as Address, // token
-              BigInt(log.topics[5] as string), //amount per second
-              this.publicClient,
-              this.walletClient
-            )
-        )
-        .forEach(handleStream);
-    };
     this.publicClient.watchContractEvent({
       address: this.address,
       abi: StreamManagerContractType.abi as Abi,
       eventName: "StreamCreated",
       args: creator ? { creator } : {},
-      onLogs,
+      onLogs: (logs: StreamCreated[]) => {
+        logs
+          .map((log) => {
+            return new Stream(
+              this,
+              log.args.creator,
+              log.args.stream_id as number,
+              log.args.token,
+              BigInt(log.args.amount_per_second),
+              this.publicClient,
+              this.walletClient
+            );
+          })
+          .forEach(handleStream);
+      },
+      onError: (error) => console.log(error),
     });
   }
+
+  async fetchAllLogs(callback: (logs: Log[]) => void) {
+    try {
+      const logs = await this.publicClient.getLogs({
+        address: this.address,
+        // TODO: pass block as function params? Otherwise takes too long
+        fromBlock: 4596186n,
```

Comment:
yeah, we have this in the python sdk, can have this be an optional parameter that defaults to block 0, and then can see if you can obtain the deployment block of the StreamManager somehow

Reply:
probably need to find some way to fetch this during the async constructor method